### PR TITLE
added some unit tests, created utility files, our sensitive keys are …

### LINKED ISSUE #4: create database
- Django
deadline: Thursday night

### PR DESCRIPTION
**Pull request by nyior**
…now hidden in env variables

### CODE REVIEW

File: apps/posts/models.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class Category(models.Model):
+	title = models.CharField(max_length=20)
+
+	def __str__(self):
+		return self.name
+
+
+class Post(models.Model):
```
Comment (diananova) [inline apps/posts/models.py:40]: shouldn't we add email and category fields here?

**Review by diananova — APPROVED** [review]
looks good to me 👍

File: apps/core/utils.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        if character == " ":
+            alphanumeric += "-"
+
+        if character == "-":
+            pass
+
+        if character.isalnum():
+            alphanumeric += character
+
+    return alphanumeric + generate_random_string()
```
Comment (gmelodie) [inline apps/core/utils.py:35]: Remove this function and just use Python's [`replace()`](https://www.w3schools.com/python/ref_string_replace.asp):

```python
some_str = "hello there"
some_str.replace(" ", "-") # replaces spaces with dashes
```

File: apps/posts/admin.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from django.contrib import admin
+from apps.posts.models import (
+                                Post, 
+                                Category
+                                )
```
Comment (gmelodie) [inline apps/posts/admin.py]:
```suggestion
from apps.posts.models import Post
from apps.posts.models import Category
```

File: apps/posts/api/pagination.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		The paginator instance associated with the view, or `None`.
+		"""
+		if not hasattr(self, '_paginator'):
+			if self.pagination_class is None:
+				self._paginator = None
+			else:
+				self._paginator = self.pagination_class()
+		else:
+			pass
+		return self._paginator
```
Comment (gmelodie) [inline apps/posts/api/pagination.py]:
```suggestion
if not hasattr(self, '_paginator'):
    self._paginator = None
    if self.pagination_class is not None:
        self._paginator = self.pagination_class()
return self._paginator
```

File: apps/posts/api/urls.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from django.urls import path, include
+
+
+
+urlpatterns = [
+    
+]
```
Comment (gmelodie) [inline apps/posts/api/urls.py:7]: Is this a mistake?

File: apps/posts/models.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    def __str__(self):
+        return self.name
+
+
+class Country(models.Model):
+    """this models a country"""
+    name = models.CharField(
+                            max_length=100, 
+                            unique=True
+                            )
```
Comment (gmelodie) [inline apps/posts/models.py]:
```suggestion
    name = models.CharField(max_length=100, unique=True)
```

**Review by gmelodie — CHANGES_REQUESTED** [review]
Love it! Just a few remarks.

File: apps/posts/models.py (hunk 1 above)
Comment (nyior) [reply apps/posts/models.py:40]: Oh I added those in the second or I opened

File: apps/core/utils.py (hunk 2 above)
Comment (nyior) [reply apps/core/utils.py:35]: @gmelodie thanks the .replace would actually make my work easier. I would need your input on this though:
Yeah, so I'm generating the slug from an item's title, and even though it's very unlikely, I imagine that some titles might have special characters. For example: "You need a Spanish dictionary?". So when generating slugs I'm not just removing white spaces, I'm also trying to remove special characters.
Any idea of how I could achieve this with the .replace?

File: apps/posts/api/urls.py (hunk 5 above)
Comment (nyior) [reply apps/posts/api/urls.py:7]: I think I've added stuff there in the second pr

**Pull request merged**